client/src/modules/list/components/Items/ListItem/ListItem.jsx

```diff
@@ -68,13 +71,32 @@ class ListItem extends PureComponent {
       return;
     }
 
-    this.setState(({ done }) => ({ done: !done }));
+    this.setState(({ done }) => ({ done: !done, disableToggleButton: true }));
 
     const shouldChangeAuthor = isNotSameAuthor && isOrdered;
 
-    return shouldChangeAuthor
-      ? toggle(isOrdered, _id, listId, userId, name)
-      : toggle(isOrdered, _id, listId);
+    /* setState method has to be called 100ms after toggle action is resolved
+     ** otherwise we will have can'tCallSetState on unmounted component error
+     */
+    if (shouldChangeAuthor) {
+      return toggle(isOrdered, _id, listId, userId, name)
+        .finally(() => {
```

Comment:
Assign the whole chain to a result variable and console log it is promise returned. 
So the promise is passed to the pending button

Reply:
Because finally returns a promise also, but these are different promises. In the pending button, you react on promise returned by finally, but you perform code from finally when toggle function returns a promise. Maybe it could be done like this:
```
const action = toggle(isOrdered, _id, listId, userId, name);
action.finally(...);
return action
```
I'm not sure if this is a better solution, but listItem and pending button react work with the same promise